httpservice/app/app.py

```diff
@@ -42,20 +44,58 @@ def create_tarfile(directory, tar_filename):
 
 @app.post("/changemaps/", dependencies=[fastapi.Security(check_apikey_header)])
 async def make_changemaps(
-    feacoll: Any = fastapi.Body(), output_tar=fastapi.Depends(sendable_tempfile)
+    data: dict = fastapi.Body(...),
+    output_tar=fastapi.Depends(sendable_tempfile)
 ):
+    input_geojson = data.get("input_geojson")
```

Comment:
You don't need to make this change here, but consider it your FastAPI lesson for the day.

Here, we silently move on even if "input_geojson" is missing.  I am of the mindset that if a precondition is violated, our code should fail loudly at the first chance you get (rather than much later in unrelated code that tries to do something with a nonexistent input_geojson). 

One (rather hasty) way to do this is by using the `data["input_geojson"]` dict accessor instead of `.get()`. That will raise KeyError if it's missing (but not if it's present but invalid), and FastAPI will convert the KeyError to a HTTP status 500.

But 500 is reserved for "something internal to the server is wrong", whereas here we'd want a 4XX, meaning "it was the caller's fault, please fix it before you try again".  The way to do that in FastAPI is introduced at https://fastapi.tiangolo.com/tutorial/body/

```python
from pydantic import BaseModel, Field

class ChangemapRequestData(BaseModel):
    input_geojson: dict
    images: dict = Field(default_factory=dict)


async def make_changemaps(
    data: ChangemapRequestData,
    output_tar=fastapi.Depends(sendable_tempfile)
):
    input_geojson = data.input_geojson
    images = data.images
    # ...
```

This is untested, but I think it pushes the request data validation into FastAPI itself (i.e. before it runs the first line of your code inside `make_changemaps`, and I also believe that FastAPI will detect that it's a data validation violation and correctly return a 4xx response.

Reply:
Cool. Can confirm that the above bit of code returns 4xx response when the data validation fails, and 200 when the inputs are as expected. Pushed the code.